packages/i18n-calypso/test/use-translate.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ShallowRenderer from 'react-test-renderer/shallow';
+
+/**
+ * Internal dependencies
+ */
+import i18n, { useTranslate } from '../src';
+
+function Label() {
+	const [ translate, lang ] = useTranslate();
+	return translate( 'hook (%(lang)s)', { args: { lang } } );
+}
+
+describe( 'useTranslate()', () => {
+	test( 'renders a translated string', () => {
+		// set some locale data
+		i18n.setLocale( {
+			'': { localeSlug: 'cs' },
+			'hook (%(lang)s)': [ 'háček (%(lang)s)' ],
+		} );
+
+		// render the Label component
+		const renderer = new ShallowRenderer();
+		renderer.render( <Label /> );
+
+		// check that it's translated
+		expect( renderer.getRenderOutput() ).toBe( 'háček (cs)' );
```

Comment:
Mocking `i18n` might allow you to test reactivity for this hook. Take a look at the `react-helpers.js` test in https://github.com/Automattic/wp-calypso/pull/31081/files for a similar situation where I mocked a browser API.